packages/xsnap-lockdown/README.md

```diff
@@ -0,0 +1,85 @@
+
+* We must evaluate a "lockdown" bundle within the worker, transforming it from plain (unsecured) JS to our preferred Endo environment. This tames the global constructors, adds the `Compartment` constructor, and removes ambient authority. It also creates a `console` object around the basic XS `print` function.
+* Then, we need to install a "supervisor" (i.e. evaluate the supervisor bundle), which hooks into the globally-registered handler function to accept delivery messages. The supervisor includes the "liveslots" code, which constructs an object-capability environment with messages routed through syscalls, as well as virtual/durable object support (through the vat-data package).
+* Finally, we'll install the actual vat bundle and run its `buildRootObject()` function.
+
+This package provides the "lockdown" bundle, which incorporates Endo and the console constructor. The bundle is generated at build time, so the contents are fixed for any given version of the `@agoric/xsnap-lockdown` package. The exported API includes a function which returns the lockdown bundle contents, as well as other helper bundles for debugging purposes.
+
+## Bundle Stability
+
+The main purpose of this `@agoric/xsnap-lockdown` package is to maintain the stability of the lockdown bundle, even when the versions of other components (XS, `xsnap`, the supervisor/liveslots bundles, or the kernel) might change. Deterministic execution is critical for a consensus/blockchain application, and a kernel upgrade should not accidentally cause the contents of the lockdown bundle to change.
+
+When the Endo source bundler is used, with `{ moduleFormat: 'endoZipBase64' }`, the generated bundles include package names and version numbers of all the source files in the transitive import graph starting from the entry point. This means the bundle contents are sensitive to version-number changes, even if the code itself remains the same.
+
+The lockdown bundle does not use this format: the initial `xsnap` environment does not yet have a loader for `endoZipBase64`-format bundles, so instead we use `nestedEvaluate`, which is fairly easy to evaluate into place. This format does not include as much metadata (like version numbers), nevertheless using a separate package for the bundle makes it easier to maintain stability of the contents.
+
+## API
+
+Bundles are JS Objects with a `moduleFormat` property (a string), and then a few other properties that depend on the module format. The `nestedEvaluate` format includes a `.source` property (a large string, suitable for IIFE evaluation) and sometimes a `.sourceMap` property. We can also JSON-serialize the bundle for storage in a single string, on disk or in a database.
+
+The primary job of this package is to provide the lockdown bundle in a form that can be delivered to the `xsnap` process for evaluation in its "Start Compartment". So the primary API is a `getLockdownBundle()`, which returns the JS Object form.
```

Comment:
Technically no, but there are tests (test-xsnap-store.js) which use a set of auxilliary bundles (`bundle-ses-boot.umd.js`, created with `rollup` instead of Endo's `bundleSource`), and I needed a way to export them too.

I'll look to see if I can change that test to use the `nestedEvaluate` form, and maybe get rid of the aux bundles, which might let us get rid of `bundlePaths`.

Reply:
Ok, I'm getting rid of `bundlePaths`, at least as a primary export. I'll leave the `paths.js` as a way for enthusiastic clients to `import { bundlePaths, entryPaths } from '@agoric/xsnap-lockdown/src/paths.js'`, and if that breaks, they get to keep both pieces.